robolectric/src/main/java/org/robolectric/internal/ParallelUniverse.java

```diff
@@ -142,13 +143,6 @@ public void setUpApplicationState(Method method, TestLifecycle testLifecycle, Re
 
   @Override
   public void tearDownApplication() {
-    // We would like to null this out (which would call reset) but that can kill async threads
-    // that run after a test complete (because they might call Context.getApplicationInfo()).
-    RobolectricPackageManager rpm = RuntimeEnvironment.getRobolectricPackageManager();
-    if (rpm != null) {
-      rpm.reset();
```

Comment:
For two reasons: 
1. It was being called twice - once here and again from resetStaticState() (Robolectric.reset() -,> RuntimeEnvironment.setRobobolectricPackageManager(). Didn't see the need to call it twice.
2. As noted, the whole point of this PR is to make that reset call after the shadows are reset. because resetStaticState() comes after tearDownApplication() so i removed the reset call there.

If you've got better ideas I'm open! :smile: